public/app/features/annotations/standardAnnotationSupport.ts

```diff
@@ -226,3 +226,21 @@ export function getAnnotationsFromData(
     })
   );
 }
+
+// These opt outs are here only for quicker and easier migration to react based annotations editors and because
+// annotation support API needs some work to support less "standard" editors like prometheus and here it is not
+// polluting public API.
+
+/**
+ * Opt out of using the default mapping functionality on frontend.
+ */
+export function shouldUseMappingUI(datasource: DataSourceApi): boolean {
+  return datasource.type !== 'prometheus';
```

Comment:
Possibly. Depends on whether data source can just use the defaults (which hopefully most can) or need to opt out. I know that putting here names of external data sources is not great but this is in order to not bloat the public API at this point so we have time to rethink it later on.

Reply:
yeah, that's what i was thinking as well. maybe external data source support can be handled in a follow-up PR if this one is ready to go?

Reply:
The main thing is that I did not want to change the API. Is it a big problem if you add a name of other datasource here? I know it's not great but would like to not change the API until we have a way to actually make it better instead of just adding more switches on top of it.

Reply:
None of our data sources use the standard mapping ui, so my +1 for extending the api so that one can opt out on the mapping ui. It doesn't have to be a breaking change?

Reply:
No but the question is, if it's in public API people will start using it (outside grafana) so it will be harder to change. Argument could be made that we will break it either way eventually. If people feel strongly I can revert to previous version where these were part of the AnnotationSupport interface

Reply:
> we can accept custom field mapping UI.
not in scope btw for g9 :)

Reply:
@aocenas i get what you mean about the public API, and agree it's not ideal to change that if there's another way to avoid it.

>  Is it a big problem if you add a name of other datasource here?

it seems like the potential issues with relying on names could be:
* external data sources are not tied to grafana versions like core data sources, so it would be a lot easier if they could opt in to this behavior as needed without waiting for a Grafana release
* would we add community & private plugins? it seems like if our plugins need this, then others would as well

either way, this seems like something that could be handled separately from this PR unless it would significantly change the approach for prometheus

Reply:
@toddtreece here is how it looked before: 
https://github.com/grafana/grafana/pull/48814/commits/11532df04ba0d48daf7840bffae6a1b6ebd749ff#diff-e0383d43bc48bbb46542917618e4ed6c5317199015f5763b22845ee57f71fbe7R117

So yeah this would solve both of those issues. As said don't like band-aiding this but may be better still for the short term.

Reply:
mentioned this in slack, but bubbling up to the discussion in the PR. i think it's probably not worth holding up this PR for any changes to the public API, i just wanted to bring it up in case it could be an issue. it seems like something that can be discussed and solved separately. 

since they are optional props, could they be introduced in future minor releases if needed? maybe that will buy extra time?